cpp/monitoring/prometheus/gauge_test.cc

```diff
@@ -0,0 +1,111 @@
+#include "monitoring/monitoring.h"
+
+#include <glog/logging.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <memory>
+
+#include "util/testing.h"
+
+namespace cert_trans {
+
+using std::string;
+using std::vector;
+using testing::ElementsAre;
+
+class GaugeTest : public ::testing::Test {
+ protected:
+  template <class... LabelTypes>
+  const string& GetName(const Metric<LabelTypes...>& m) {
+    return m.Name();
+  }
+
+  template <class... LabelTypes>
+  const vector<string>& GetLabelNames(const Metric<LabelTypes...>& m) {
+    return m.LabelNames();
+  }
+
+  template <class... LabelTypes>
+  string GetHelp(const Metric<LabelTypes...>& m) {
+    return m.Help();
+  }
+};
+
+
+TEST_F(GaugeTest, TestGaugeName) {
+  std::unique_ptr<PrometheusGauge<>> gauge(
+      PrometheusGauge<>::New("name", "help"));
+  EXPECT_EQ("name", GetName(*gauge));
+}
+
+
+TEST_F(GaugeTest, TestGaugeLabelNamesEmpty) {
+  std::unique_ptr<PrometheusGauge<>> gauge(
+      PrometheusGauge<>::New("name", "help"));
+  EXPECT_TRUE(GetLabelNames(*gauge).empty());
+}
+
+
+TEST_F(GaugeTest, TestGaugeLabelNames) {
+  std::unique_ptr<PrometheusGauge<string, string, int>> gauge(
+      PrometheusGauge<string, string, int>::New("name", "one", "two", "three",
+                                                "help"));
+  EXPECT_THAT(GetLabelNames(*gauge), ElementsAre("one", "two", "three"));
+}
+
+TEST_F(GaugeTest, TestGaugeHelp) {
+  std::unique_ptr<PrometheusGauge<>> gauge(
+      PrometheusGauge<>::New("name", "help"));
+  EXPECT_EQ("help", GetHelp(*gauge));
+}
+
+
+TEST_F(GaugeTest, TestGauge) {
+  std::unique_ptr<PrometheusGauge<>> gauge(
+      PrometheusGauge<>::New("name", "help"));
+  EXPECT_EQ(0, gauge->Get());
+  gauge->Set(15);
+  EXPECT_EQ(15, gauge->Get());
+  gauge->Set(1);
+  EXPECT_EQ(1, gauge->Get());
+}
+
+
+TEST_F(GaugeTest, TestGaugeWithLabels) {
+  std::unique_ptr<PrometheusGauge<std::string, int>> gauge(
+      PrometheusGauge<std::string, int>::New("name", string("a string"),
```

Comment:
Try removing those ```string()```? Maybe it'll just work? Maybe you can make the typedef in ```NameType``` a ```const char*``` instead? Or is there a new special C++11 type to say "literal string"? It might be only where you have the literal, rather than where you take it...